x-pack/plugins/cloud_security_posture/public/common/api/create_detection_rule.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { HttpSetup } from '@kbn/core/public';
+
+const DETECTION_ENGINE_URL = '/api/detection_engine' as const;
+const DETECTION_ENGINE_RULES_URL = `${DETECTION_ENGINE_URL}/rules` as const;
+
+interface RuleCreateProps {
+  type: string;
+  language: string;
+  license: string;
+  author: string[];
+  filters: any[];
+  false_positives: any[];
+  risk_score: number;
+  risk_score_mapping: any[];
+  severity: string;
+  severity_mapping: any[];
+  threat: any[];
+  interval: string;
+  from: string;
+  to: string;
+  timestamp_override: string;
+  timestamp_override_fallback_disabled: boolean;
+  actions: any[];
+  enabled: boolean;
+  alert_suppression: {
+    group_by: string[];
+    missing_fields_strategy: string;
+  };
+  index: string[];
+  query: string;
+  references: string[];
+  name: string;
+  description: string;
+  tags: string[];
+}
+
+export interface RuleResponse extends RuleCreateProps {
+  id: string;
+}
+
+/**
+ * Creates a detection rule
+ *
+ */
+export async function createDetectionRule({
+  http,
+  rule,
+}: {
+  http: HttpSetup;
+  rule: RuleCreateProps;
+}): Promise<RuleResponse> {
+  const res = await http.post<RuleCreateProps>(DETECTION_ENGINE_RULES_URL, {
+    body: JSON.stringify(rule),
+  });
+
+  return res as RuleResponse;
```

Comment:
yes, that will be a nice enhancement, but we would still need this HTTP post function, as react-query doesn't handle the HTTP calls itself since it's mainly a caching and state management library for API calls. 

Also, since this is a POST method we would need to use react query's useMutation like this: 

https://github.com/elastic/kibana/blob/main/x-pack/plugins/cases/public/containers/use_bulk_update_case.tsx

But as we don't have the requirement to use the caching just yet I ended up using only the fetch call, but I'll wrap the calls on `react-query` once I manage to work on [this](https://github.com/elastic/kibana/issues/162109) task so we can access the `queryClient.invalidateQueries();` to `refetch` the rules once a rule is created.